src/string_object.cpp

```diff
@@ -2256,6 +2256,38 @@ Value StringObject::upcase(Env *env) {
     return str;
 }
 
+Value StringObject::capitalize(Env *env) {
+    auto str = new StringObject { "", m_encoding };
+    if (length() == 0) {
+        return str;
+    } else {
+        auto ary = chars(env)->as_array();
```

Comment:
Splitting the string into chars is a very expensive operation. I wonder if there's something we can do with the bytes in-place. And I agree with @herwinw we should do the bang method first and use `dup.upcase!` for the non-bang method.

Reply:
I'll poke around some more in the codebase in search of a different solution and get to work on a bang implementation. If you would like to close this in favor of a separate PR down the line, that's entirely fine with me. I'm not well-versed with git, so I don't know if this one can be modified.

Reply:
So I'm thinking we could call `next_char(0)` to just get the first character. That would be way faster.